### ISSUE TITLE
kola: rawhide: 20221021: new NetworkManager 1.41.3 breaks network tests

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#1325]
The `ext.config.networking.default-network-behavior-change` and `ext.config.networking.mtu-on-bond-ignition` tests both fail with the new NetworkManager that got pulled in recently:

```
NetworkManager 1:1.40.0-1.fc38 -> 1:1.41.3-1.fc38
```

Let's pin and investigate.

Failure output:

<details>

```
[2022-10-21T21:46:51.534Z] ======== Re-running failed tests (flake detection) ========
[2022-10-21T21:46:51.534Z] 
[2022-10-21T21:46:51.534Z] ⚠️  Skipping kola test pattern "fcos.internet":
[2022-10-21T21:46:51.534Z]   👉 https://github.com/coreos/coreos-assembler/pull/1478
[2022-10-21T21:46:51.534Z] ⚠️  Skipping kola test pattern "podman.workflow":
[2022-10-21T21:46:51.534Z]   👉 https://github.com/coreos/coreos-assembler/pull/1478
[2022-10-21T21:46:51.534Z] === RUN   ext.config.networking.mtu-on-bond-ignition
[2022-10-21T21:46:51.534Z] === RUN   non-exclusive-test-bucket-0
[2022-10-21T21:47:06.340Z] === RUN   non-exclusive-test-bucket-0/ext.config.networking.default-network-behavior-change
[2022-10-21T21:47:08.214Z] systemctl status kola-runext-42.service:
[2022-10-21T21:47:08.214Z] × kola-runext-42.service
[2022-10-21T21:47:08.214Z]      Loaded: loaded (/etc/systemd/system/kola-runext-42.service; static)
[2022-10-21T21:47:08.214Z]      Active: failed (Result: exit-code) since Fri 2022-10-21 21:47:06 UTC; 1s ago
[2022-10-21T21:47:08.214Z]    Duration: 39ms
[2022-10-21T21:47:08.214Z]     Process: 1887 ExecStart=/usr/local/bin/kola-runext-test.sh (code=exited, status=1/FAILURE)
[2022-10-21T21:47:08.214Z]    Main PID: 1887 (code=exited, status=1/FAILURE)
[2022-10-21T21:47:08.214Z]         CPU: 15ms
[2022-10-21T21:47:08.214Z] 
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1893]: [ipv6]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1893]: dhcp-timeout=90
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1893]: method=auto
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1893]: [proxy]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1893]: [user]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1893]: org.freedesktop.NetworkManager.origin=nm-initrd-generator'
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: ++ echo '# Created by nm-initrd-generator
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: [connection]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: id=Wired Connection
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: uuid=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: type=ethernet
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: autoconnect-retries=1
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: multi-connect=3
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: [ethernet]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: [ipv4]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: dhcp-timeout=90
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: method=auto
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: required-timeout=20000
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: [ipv6]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: dhcp-timeout=90
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: method=auto
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: [proxy]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: [user]
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1892]: org.freedesktop.NetworkManager.origin=nm-initrd-generator'
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]: --- /dev/fd/63        2022-10-21 21:47:06.541427436 +0000
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]: +++ /dev/fd/62        2022-10-21 21:47:06.542427436 +0000
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]: @@ -4,6 +4,7 @@
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]:  id=Wired Connection
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]:  uuid=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]:  type=ethernet
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]: +autoconnect-priority=-100
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]:  autoconnect-retries=1
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]:  multi-connect=3
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1894]:  
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1887]: + fatal 'fail: the expected initrd network config is not given by the kargs'
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1887]: + echo 'fail: the expected initrd network config is not given by the kargs'
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1887]: fail: the expected initrd network config is not given by the kargs
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost kola-runext-test.sh[1887]: + exit 1
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost systemd[1]: kola-runext-42.service: Main process exited, code=exited, status=1/FAILURE
[2022-10-21T21:47:08.214Z] Oct 21 21:47:06 localhost systemd[1]: kola-runext-42.service: Failed with result 'exit-code'.
[2022-10-21T21:47:09.568Z] systemctl status kola-runext.service:
[2022-10-21T21:47:09.568Z] × kola-runext.service
[2022-10-21T21:47:09.568Z]      Loaded: loaded (/etc/systemd/system/kola-runext.service; static)
[2022-10-21T21:47:09.568Z]      Active: failed (Result: exit-code) since Fri 2022-10-21 21:47:08 UTC; 1s ago
[2022-10-21T21:47:09.568Z]    Duration: 35ms
[2022-10-21T21:47:09.568Z]     Process: 1931 ExecStart=/usr/local/bin/kola-runext-test.sh (code=exited, status=10)
[2022-10-21T21:47:09.568Z]    Main PID: 1931 (code=exited, status=10)
[2022-10-21T21:47:09.568Z]         CPU: 15ms
[2022-10-21T21:47:09.568Z] 
[2022-10-21T21:47:09.568Z] Oct 21 21:47:08 localhost systemd[1]: Started kola-runext.service.
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1931]: ++ cmdline=($(< /proc/cmdline))
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1931]: + bond=bond0
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1931]: + vlan=bond0.100
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1931]: + for interface in $bond $vlan
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1931]: + mtu=
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1932]: ++ nmcli -g 802-3-ethernet.mtu connection show bond0
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1932]: Error: bond0 - no such connection profile.
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost kola-runext-test.sh[1931]: + mtu=
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost systemd[1]: kola-runext.service: Main process exited, code=exited, status=10/n/a
[2022-10-21T21:47:09.569Z] Oct 21 21:47:08 localhost systemd[1]: kola-runext.service: Failed with result 'exit-code'.
[2022-10-21T21:47:10.124Z] --- FAIL: non-exclusive-test-bucket-0 (27.35s)
[2022-10-21T21:47:10.124Z]     --- FAIL: non-exclusive-test-bucket-0/ext.config.networking.default-network-behavior-change (1.60s)
[2022-10-21T21:47:10.124Z]             cluster.go:162: Error: Unit kola-runext-42.service exited with code 1
[2022-10-21T21:47:10.124Z]             cluster.go:162: 2022-10-21T21:47:07Z cli: Unit kola-runext-42.service exited with code 1
[2022-10-21T21:47:10.124Z]             harness.go:1093: kolet failed: : kolet run-test-unit failed: Process exited with status 1
[2022-10-21T21:47:11.475Z] --- FAIL: ext.config.networking.mtu-on-bond-ignition (28.72s)
[2022-10-21T21:47:11.475Z]         cluster.go:162: Error: Unit kola-runext.service exited with code 10
[2022-10-21T21:47:11.475Z]         cluster.go:162: 2022-10-21T21:47:09Z cli: Unit kola-runext.service exited with code 10
[2022-10-21T21:47:11.475Z]         harness.go:1093: kolet failed: : kolet run-test-unit failed: Process exited with status 1
[2022-10-21T21:47:11.475Z] FAIL, output in /home/jenkins/agent/workspace/build/tmp/kola-F14jZ/kola/rerun
[2022-10-21T21:47:11.475Z] Error: harness: test suite failed
```

</details>

**Comment by dustymabe** [comment]
Pin PR in https://github.com/coreos/fedora-coreos-config/pull/2029

**Comment by HuijingHei** [comment]
For `ext.config.networking.mtu-on-bond-ignition`, find the igntion config does not make effect, will dig more

```
$ cosa run --additional-nics 2 --kargs "net.ifnames=0" -m 4096 -B src/config/tests/kola/shared/networking/mtu-on-bond-ignition/config.bu

[root@cosa-devsh ~]# rpm -q NetworkManager
NetworkManager-1.41.3-1.fc38.x86_64
# nmcli c show | more
NAME                UUID                                  TYPE      DEVICE 
Wired connection 1  177cc262-14f6-3385-a138-05c6c290d03a  ethernet  eth0   
Wired connection 2  c4d4a587-4c6d-3033-b6fa-b7897c3b031b  ethernet  eth1   
eth2                b07c334a-399b-32de-8d50-58e4e08f98e3  ethernet  -- 
```

**Comment by dustymabe** [comment]
For the `ext.config.networking.mtu-on-bond-ignition` test I opened: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/1130

**Comment by dustymabe** [comment]
For `ext.config.networking.default-network-behavior-change` the first commit in https://github.com/coreos/fedora-coreos-config/pull/2058 should address it.

While we wait on a fix for `ext.config.networking.mtu-on-bond-ignition` we can snooze the test for `rawhide` (last commit in https://github.com/coreos/fedora-coreos-config/pull/2058) and unpin the package in `rawhide` once that merges.

**Comment by HuijingHei** [comment]
Thanks @dustymabe for the digging! Both 2 cases are failed on scos with `NetworkManager-1.41.3-1.el9.x86_64`.

**Comment by dustymabe** [comment]
This is fixed upstream in [545edb70](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/545edb7093412e7ea936fc788d63d5ddc9f4e51e).

**Comment by dustymabe** [comment]
The two problems documented by this issue are now fixed:

- https://github.com/coreos/fedora-coreos-config/pull/2058
- https://github.com/coreos/fedora-coreos-config/pull/2059
